ST417 Bayesian Modelling Project.Rmd

```diff
@@ -67,49 +71,217 @@ scale_and_avg_range <- function(column, to_match, scale) {
+bayes_df_cost <- data.frame(theta_cost, prior_cost, likelihood_cost)
+
+# posterior calculation
+bayes_df_cost$Product <- bayes_df_cost$prior_cost * bayes_df_cost$likelihood_cost # nolint
+bayes_df_cost$Posterior <- bayes_df_cost$Product / sum(bayes_df_cost$Product)
+bayes_df_cost
+
+# prior and posterior comparison
+prior_post_plot(bayes_df_cost)
+```
+
+```{r}
+m0_time <- mean(data$est_time, na.rm = TRUE)
+m0_dist <- mean(data$est_distance, na.rm = TRUE)
+m0_cost <- mean(data$est_cost, na.rm = TRUE)
+s0_time <- sd(data$est_time, na.rm = TRUE)
+s0_dist <- sd(data$est_distance, na.rm = TRUE)
+s0_cost <- sd(data$est_cost, na.rm = TRUE)
+```
+
+```{r}
+post_mean_time <- ((1 / (se_time^2) + 1 / (s0_time^2))^(-1)) * (xbar_time / (se_time^2) + m0_time / (s0_time^2)) # nolint
+post_sd_time <- sqrt((1 / (se_time^2) + 1 / (s0_time^2))^(-1))
+
+post_mean_dist <- (1 / (se_dist^2) + 1 / (s0_dist^2))^(-1) * (xbar_dist / (se_dist^2) + m0_dist / (s0_dist^2)) # nolint
+post_sd_dist <- sqrt((1 / (se_dist^2) + 1 / (s0_dist^2))^(-1))
+
+post_mean_cost <- (1 / (se_cost^2) + 1 / (s0_cost^2))^(-1) * (xbar_cost / (se_cost^2) + m0_cost / (s0_cost^2)) # nolint
+post_sd_cost <- sqrt((1 / (se_cost^2) + 1 / (s0_cost^2))^(-1))
+```
+
+```{r 95 credible intervals}
+# 95% credible intervals
+time_sims <- rnorm(10000, mean = post_mean_time, sd = post_sd_time)
+time_point_estimate <- mean(time_sims)
+time_point_estimate
+
+time_ci95 <- quantile(time_sims, probs = c(0.025, 0.975))
+time_ci95
+normal_interval(0.95, time_ci95)
+
+dist_sims <- rnorm(10000, mean = post_mean_dist, sd = post_sd_dist)
+dist_point_estimate <- mean(dist_sims)
+dist_point_estimate
+
+dist_ci95 <- quantile(dist_sims, probs = c(0.025, 0.975))
+dist_ci95
+normal_interval(0.95, dist_ci95)
+
+cost_sims <- rnorm(10000, mean = post_mean_cost, sd = post_sd_cost)
+cost_point_estimate <- mean(cost_sims)
+cost_point_estimate
 
+cost_ci95 <- quantile(cost_sims, probs = c(0.025, 0.975))
+cost_ci95
+normal_interval(0.95, cost_ci95)
+```
+
+```{r distributions plot}
+x <- seq(-60, 80, length = 150)
+
+priorx_time <- dnorm(x, mean = mean(data$est_time, na.rm = TRUE),
+                     sd = sd(data$est_time, na.rm = TRUE))
+priorx_dist <- dnorm(x, mean = mean(data$est_distance, na.rm = TRUE),
+                     sd = sd(data$est_distance, na.rm = TRUE))
+priorx_cost <- dnorm(x, mean = mean(data$est_cost, na.rm = TRUE),
+                     sd = sd(data$est_cost, na.rm = TRUE))
+
+datax_time  <- dnorm(x, mean = mean(data$time, na.rm = TRUE), sd = se_time)
+datax_dist  <- dnorm(x, mean = mean(data$distance, na.rm = TRUE), sd = se_dist)
+datax_cost  <- dnorm(x, mean = mean(data$cost, na.rm = TRUE), sd = se_cost)
+
+postx_time <- dnorm(x, mean = post_mean_time, sd = post_sd_time)
+postx_dist <- dnorm(x, mean = post_mean_dist, sd = post_sd_dist)
+postx_cost <- dnorm(x, mean = post_mean_cost, sd = post_sd_cost)
+
+plot(x, priorx_time, type = "l", lwd = 3, xlim = c(-40, 60), ylim = c(0, 0.8),
```

Comment:
It's a really cool idea plotting these against each other! However, I think they're a bit muddled, and the meaning of theta is different for time, cost and distance. Could you please plot them as three subplots? 

It's worth noting that the issue of negative theta appears here also.